lib/maropost/contact.rb

```diff
@@ -0,0 +1,91 @@
+require 'active_model'
+
+module Maropost
+  class Contact
+    include ActiveModel::Model
+
+    attr_accessor :id, :email,
+      :travel_weekly, :enews, :travel_especials, :ins_enews, :deals_discounts, :new_member_series, :fleet_contact,
+      :personal_vehicle_reminder, :business_vehicle_reminder, :associate_vehicle_reminder,
+      :vr_reminder_email, :vr_reminder_sms, :vr_reminder_autocall
+
+    def self.find(email)
+      response = Maropost::Api.new(data: { email: email }).find
+      contact = JSON.parse response
+      build contact
+    rescue RestClient::ResourceNotFound => e
+      new(email: email).tap do |contact|
+        contact.errors.add :base, "Email not found"
+      end
+    end
+
+    def create_or_update
+      api = Maropost::Api.new(data: contact_data)
+      response = exists? ? api.update : api.create
+      contact = JSON.parse response
+      self.class.build contact
```

Comment:
Should `create_or_update` be a class method too? Might match AR's i.e. `Foo.create` vs `Foo.new.create`